packages/spatial/src/input/systems/ClientInputSystem.tsx

```diff
@@ -291,7 +285,7 @@ const execute = () => {
       let closestDistanceSquared = Infinity
 
       //use sourceEid if controller (one InputSource per controller), otherwise use avatar rather than InputSource-emulated-pointer
-      const selfAvatarEntity = AvatarComponent.getSelfAvatarEntity()
+      const selfAvatarEntity = UUIDComponent.getEntityByUUID((Engine.instance.userID + '_avatar') as EntityUUID) //would prefer a better way to do this
```

Comment:
For a future reference, one strategy (which I think @HexaField  suggested at some point), would be to inject these heuristics from higher level packages, so perhaps the `world` (currently `engine`) package injects the additional proximity heuristic.

Reply:
I would love to learn more, not sure from the description what that would look like

Reply:
why does this system care about the avatar? what is the early scape in the for loop doing?

Reply:
Because one of the heuristics for focusing non-spatial input sources on entities that can receive input is some notion of proximity to those entities (e.g., when not pointing w/ the mouse cursor)